hs_composite_resource/tests/test_composite_resource_user_zone.py

```diff
@@ -181,13 +953,41 @@ def test_delete_composite_resource(self):
 
         self.assertEqual(BaseResource.objects.count(), 0)
         self._create_composite_resource()
-        # set the logical file - which get sets as part of the post resource creation signal
-        resource_post_create_actions(resource=self.composite_resource, user=self.user,
-                                     metadata=self.composite_resource.metadata)
-
+        res_file = self.composite_resource.files.first()
+        # create the generic aggregation (logical file)
+        GenericLogicalFile.set_file_type(self.composite_resource, self.user, res_file.id)
+        self.assertEqual(GenericLogicalFile.objects.count(), 1)
         self.assertEqual(BaseResource.objects.count(), 1)
         self.composite_resource.delete()
         self.assertEqual(BaseResource.objects.count(), 0)
+        self.assertEqual(GenericLogicalFile.objects.count(), 0)
+
+    def test_copy_composite_resource_with_single_file_aggregation(self):
+        # test that we can create a copy of a composite resource that has a single file
+        # aggregation
+
+        self._test_copy_or_version_of_composite_resource(aggr_type=GenericLogicalFile,
```

Comment:
+1 for reusing test cases with different parameters.  There is an official way to do this with parameterized unit testing.  That could be useful if we want to test all the aggregation types.